kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -74,15 +75,30 @@
 })
 // @formatter:on
 public class KafkaProxyReconciler implements
-        Reconciler<KafkaProxy> {
+        Reconciler<KafkaProxy>,
+        ContextInitializer<KafkaProxy> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(KafkaProxyReconciler.class);
 
     public static final String CONFIG_DEP = "config";
     public static final String DEPLOYMENT_DEP = "deployment";
     public static final String CLUSTERS_DEP = "clusters";
+    static final String SEC = "sec";
+    private final SecureConfigInterpolator secureConfigInterpolator;
 
-    public KafkaProxyReconciler() {
+    public KafkaProxyReconciler(SecureConfigInterpolator secureConfigInterpolator) {
+        this.secureConfigInterpolator = secureConfigInterpolator;
+    }
+
+    static SecureConfigInterpolator secureConfigInterpolator(Context<KafkaProxy> context) {
```

Comment:
I'd move this static to SecureConfigInterpolator

Reply:
I disagree on this. A SecureConfigInterpolator doesn't need to know about Contexts. i.e. I see the `SecureConfigInterpolator` as a lower-level thing and how its used by Reconcilers and DependentResources is a higher level concern. 

In any case, I see this as related to [my other PR](https://github.com/kroxylicious/kroxylicious/pull/1975/files#r2007870446), and I'd prefer to tidy up this minor detail once we have all the `Reconcilers` and can understand the variety of ways we're actually using the `managedWorkflowAndDependentResourceContext`.